core/server/api/index.js

```diff
@@ -139,6 +140,10 @@ locationHeader = function locationHeader(req, result) {
         } else if (result.hasOwnProperty('tags')) {
             newObject = result.tags[0];
             location = utils.url.urlJoin(apiRoot, 'tags', newObject.id, '/');
+        } else if (result.hasOwnProperty('webhooks')) {
+            newObject = result.webhooks[0];
+            // TODO: this URL doesn't actually exist but is needed for a 201 response?
+            location = utils.url.urlJoin(apiRoot, 'webhooks', newObject.id, '/');
```

Comment:
We only [return a 201 if `location` is set](https://github.com/TryGhost/Ghost/blob/master/core/server/api/index.js#L188-L196), does it matter that the URL for set here will return a 404?

Reply:
don't we need a read/browse endpoint for webhooks anyway?

Reply:
I'm not too sure at the moment, I've implemented the minimum necessary for Zapier integration, if we want to offer read/browse we will need to consider future use-cases and how security will work with external systems being given permission to read/write/browse webhooks that weren't created by that system.

Reply:
okay i thought we want to show existing hooks in the admin.

if we need read/browse, i think as long as we don't expose the id of existing webhooks when reading/browsing, it's fine. not sure the target url itself has to be secured.

in general regarding security, we might need confidential clients. so each external service get's it's own private client creds by entering them during the app flow (instead of requesting the public client). and a client is connected to a webhook. (webhook.client_id). the differentiation should be pretty easy, but yeah future thing.

Reply:
> okay i thought we want to show existing hooks in the admin.

Not at this point, maybe later once there is more general hook support outside of `subscriber.created` and `subscriber.deleted`. I think it will also be dependent on the rethink of our API authentication so that services registering webhooks can have their own clients rather than piggy backing on the password grant flow - again something that is fine for the Zapier integration but needs more thought/design before releasing for general use.

Reply:
yeah 👍